devices/tuya.js

```diff
@@ -1969,40 +1969,89 @@ module.exports = [
+            exposes.composite('schedule', 'schedule').withFeature(exposes.enum('week_day', ea.STATE_SET, ['monday', 'tuesday',
+                'wednesday', 'thursday', 'friday', 'saturday', 'sunday'])).withFeature(exposes.text('schedule', ea.STATE_SET))
+                .withDescription('Schedule will work with "auto" preset. In this mode, the device executes ' +
+                'a preset week programming temperature time and temperature. Before using these properties, check `working_day` ' +
+                'property. Each day can contain up to 10 segments. At least 1 segment should be defined. Different count of segments ' +
+                'can be defined for each day, e.g., 3 segments for Monday, 5 segments for Thursday, etc. It should be defined in the ' +
+                'following format: `hours:minutes/temperature`. Minutes can be only tens, i.e., 00, 10, 20, 30, 40, 50. Segments should ' +
+                'be divided by space symbol. Examples: `04:00/20 08:30/22 10:10/18 18:40/24 22:50/19.5`; `06:00/21.5 17:20/26 24:00/18`. ' +
+                'The temperature will be set from the beginning/start of one period and until the next period, e.g., `04:00/20 24:00/22` ' +
+                'means that from 00:00 to 04:00 temperature will be 20 degrees and from 04:00 to 00:00 temperature will be 22 degrees.'),
             exposes.text('schedule_monday', ea.STATE),
             exposes.text('schedule_tuesday', ea.STATE),
             exposes.text('schedule_wednesday', ea.STATE),
             exposes.text('schedule_thursday', ea.STATE),
             exposes.text('schedule_friday', ea.STATE),
             exposes.text('schedule_saturday', ea.STATE),
             exposes.text('schedule_sunday', ea.STATE),
-            exposes.binary('online', ea.STATE_SET, 'ON', 'OFF').withDescription('Is the device online'),
+            exposes.binary('online', ea.STATE_SET, 'ON', 'OFF').withDescription('Turn on this property to poll current data from the ' +
+                'device. It can be used to periodically fetch a new local temperature since the device doesn\'t update itself. ' +
+                'Setting this property doesn\'t turn on the display.'),
             exposes.numeric('error_status', ea.STATE).withDescription('Error status'),
         ],
+        meta: {
+            tuyaDatapoints: [
+                [2, 'preset', tuya.valueConverterBasic.lookup({'auto': 0, 'manual': 1, 'holiday': 3}),
+                    {datatype: tuya.sendDataPointEnum}],
+                [8, 'open_window', tuya.valueConverter.onOff],
+                [10, 'frost_protection', tuya.valueConverter.onOff],
+                [16, 'current_heating_setpoint', tuya.valueConverter.divideBy10],
+                [27, 'local_temperature_calibration', tuya.valueConverter.localTempCalibration],
+                [31, 'working_day', tuya.valueConverterBasic.lookup({'mon_sun': 0, 'mon_fri+sat+sun': 1, 'separate': 2}),
+                    {datatype: tuya.sendDataPointEnum}],
```

Comment:
Instead of defining the dataType in meta I suggest the following.

- In `lib/tuya.js` define a class called `Enum`.
- Replace e.g. `'mon_sun': 0` with `'mon_sun': tuya.Enum(0)`
- In `toZigbee.js` do a check like `else if (typeof convertedValue === 'Enum') {` then use `tuya.sendDataPointEnum`

Reply:
I have made changes according to your suggestions. Review, please.